cpp/util/etcd.cc

```diff
@@ -688,14 +726,24 @@ void EtcdClient::FetchDone(RequestState* etcd_req, Task* task) {
   VLOG(2) << "EtcdClient::FetchDone: " << task->status();
 
   if (!task->status().ok()) {
-    // TODO(pphaneuf): If there is a connection problem, we should re-do the
-    // request on another node.
+    if (task->status().error_code() == util::error::UNAVAILABLE) {
+      // Seems etcd wasn't available; pick a new etcd server and retry
+      LOG(WARNING) << "Etcd fetch failed: " << task->status() << ", retrying "
+                   << "on next etcd server.";
+      ChooseNextServer();
+      etcd_req->SetHostPort(endpoint_);
+      fetcher_->Fetch(etcd_req->req_, &etcd_req->resp_,
```

Comment:
No delay, ever? It's a bit complicated to Do It Right (tm), needing things like knowing when you've gone through the whole list (so keeping track of what server you've tried)... Maybe this is sufficient, but that's the main reason I hadn't tackled this yet (didn't want to add an infinite busy loop here!).